lib/Zonemaster/Backend/TestAgent.pm

```diff
@@ -172,7 +172,7 @@ sub run {
 
     $self->{_db}->test_results( $test_id, Zonemaster::Engine->logger->json( 'INFO' ) );
 
-    $progress = $self->{_db}->test_progress( $test_id );
+    $self->{_db}->test_progress( $test_id );
```

Comment:
This line is not necessary, the only thing it does is returning the progress, you should be able to remove it safely.

Reply:
good point, I've removed it.